test/e2e/kubectl.go

```diff
@@ -1630,22 +1629,41 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 })
 
 // Checks whether the output split by line contains the required elements.
-func checkOutput(output string, required [][]string) {
+func checkOutputReturnError(output string, required [][]string) error {
 	outputLines := strings.Split(output, "\n")
 	currentLine := 0
 	for _, requirement := range required {
 		for currentLine < len(outputLines) && !strings.Contains(outputLines[currentLine], requirement[0]) {
 			currentLine++
 		}
 		if currentLine == len(outputLines) {
-			framework.Failf("Failed to find %s in %s", requirement[0], output)
+			return fmt.Errorf("failed to find %s in %s", requirement[0], output)
 		}
 		for _, item := range requirement[1:] {
 			if !strings.Contains(outputLines[currentLine], item) {
-				framework.Failf("Failed to find %s in %s", item, outputLines[currentLine])
+				return fmt.Errorf("failed to find %s in %s", item, outputLines[currentLine])
 			}
 		}
 	}
+	return nil
+}
+
+func checkOutput(output string, required [][]string) {
+	err := checkOutputReturnError(output, required)
+	if err != nil {
+		framework.Failf("%v", err)
+	}
+}
+
+func checkKubectlOutputWithRetry(required [][]string, args ...string) {
+	for i := 0; i < 5; i++ {
+		output := framework.RunKubectlOrDie(args...)
+		err := checkOutputReturnError(output, required)
+		if err != nil {
+			framework.Failf("%v", err)
```

Comment:
I got your point. We should use Errorf and PollImmediate here.
What should the polling interval be?